src/show/components/TextField/index.js

```diff
@@ -1,23 +1,83 @@
-import React from "react";
-
+import React, { useState } from 'react';
+import { ReactComponent as EyeIcon } from '../../assets/svg/eye-icon.svg';
 const TextField = ({
+  name,
+  type = 'text',
   label = "",
   placeholder = "",
-  required = false,
   value = "",
-  password = false,
-}) => (
-  <div>
-    <h6>{label}</h6>
-    <input
-      placeholder={placeholder}
-      type={password ? "password" : "text"}
-      password={true}
-      className="bg-gray-100 text-black w-64 px-2 rounded border-2 border-black"
-    //   value={value}
-    />
-    <h6 className="text-xs text-right">{required && "required"}</h6>
-  </div>
-);
+  onChange,
+  hasEye,
+}) => {
+  const [error, setError] = useState('');
+  const [inputType, setInputType] = useState(type);
+
+  const validate = (type, value) => {
+    const validationRules = {
+    email: {
+    pattern: /^[^\s@]+@[^\s@]+.[^\s@]+$/,
+    errorMessage: 'Invalid email address'
+    },
+    password: {
+    pattern: /^(?=.[0-9])(?=.[!@#$%^&])[A-Za-z\d!@#$%^&]{8,}$/,
+    errorMessage: 'Password must be at least 8 characters long and include at least one number and one special character'
+    }
+    };
+    
+    // Check if the type exists in the validation rules
+    if (validationRules[type]) {
+    const { pattern, errorMessage } = validationRules[type];
+    if (!pattern.test(value)) {
+    setError(errorMessage);
+    return;
+    }
+    }
+    
+    setError('');
+    };
+
+  const handleChange = (e) => {
+    const { value } = e.target;
+    validate(value);
+    onChange(value);
+  };
+
+  const togglePasswordVisibility = () => {
+    setInputType(inputType === 'password' ? 'text' : 'password');
+  };
+
+  return (
+    <div className="relative mb-4">
+      {label && (
+        <label className="block mb-1 text-xs font-medium text-zinc-400">
+          {label}
+        </label>
+      )}
+      <input
+        type={inputType}
+        name={name}
+        placeholder={placeholder}
+        value={value}
+        onChange={handleChange}
+        className={`w-full border rounded-md text-xs p-1 ${error ? 'border-red-500' : 'border-gray-300'} focus:outline-none`}
+      />
+      {hasEye && type === 'password' && (
```

Comment:
why we need this extra hasEye props, if the type is given password, then it should automatically take that